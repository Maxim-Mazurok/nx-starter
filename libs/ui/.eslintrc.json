{
  "extends": ["../../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.{js,jsx,ts,tsx}"],
      "extends": ["plugin:@nrwl/nx/react", "../../.eslintrc.react.js"]
    }
  ]
}