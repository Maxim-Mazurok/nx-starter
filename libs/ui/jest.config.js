module.exports = {
  displayName: 'ui',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\\.tsx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js'],
  coverageDirectory: '../../coverage/libs/ui',
};
