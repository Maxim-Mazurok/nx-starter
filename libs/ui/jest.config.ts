module.exports = {
  displayName: 'ui',
  preset: '../../tools/jest/preset.js',
  transform: {
    '^.+\\.tsx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js'],
  coverageDirectory: '../../../out/coverage/libs/ui',
  setupFilesAfterEnv: ['../../tools/jest/setups/dom.ts'],
};
