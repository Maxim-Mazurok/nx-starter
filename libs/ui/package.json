{
  "name": "@nx-starter/ui",
  "private": false,
  "version": "0.0.1",
  "dependencies": {
    "@nrwl/storybook": "^13.10.2",
    "webpack": "^5.27.2"
  },
  "devDependencies": {
    "@storybook/addon-a11y": "^6.3.8",
    "@storybook/addon-actions": "^6.3.12",
    "@storybook/addon-docs": "^6.3.8",
    "@storybook/addon-essentials": "^6.3.8",
    "@storybook/addon-links": "^6.3.8",
    "@storybook/addon-postcss": "^2.0.0",
    "@storybook/core": "^6.4.22",
    "@storybook/react": "^6.3.8",
    "@storybook/theming": "^6.3.8"
  }
}
