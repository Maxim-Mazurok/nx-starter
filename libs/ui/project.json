{
  "root": "libs/ui",
  "sourceRoot": "libs/ui/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "type-check": {
      "executor": "./tools/executors/type-check:type-check"
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["libs/ui/**/*.{js,jsx,ts,tsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "options": {
        "jestConfig": "libs/ui/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "build": {
      "executor": "@nrwl/web:rollup",
      "options": {
        "outputPath": "out/build/libs/ui",
        "tsConfig": "libs/ui/tsconfig.lib.json",
        "project": "libs/ui/package.json",
        "entryFile": "libs/ui/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": "@nrwl/react/plugins/bundle-rollup",
        "compiler": "babel",
        "format": ["esm"],
        "assets": [
          {
            "glob": "libs/ui/README.md",
            "input": ".",
            "output": "."
          }
        ]
      }
    }
  }
}
