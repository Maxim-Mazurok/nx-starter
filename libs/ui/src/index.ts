export * from './lib/ui';
export * from './utils';
