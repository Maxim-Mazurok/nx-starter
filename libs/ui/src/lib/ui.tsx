import styled from 'styled-components';

/* eslint-disable-next-line */
export interface UiProps {}

const StyledUi = styled.div`
  color: pink;
`;

export function Ui() {
  return (
    <StyledUi data-testid='idid'>
      <h1>Welcome to Ui!</h1>
    </StyledUi>
  );
}
