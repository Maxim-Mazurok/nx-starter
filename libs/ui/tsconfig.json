{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "rootDir": ".",
    "jsx": "react-jsx",
    "jsxImportSource": "@emotion/react"
  },
  "references": [
    { "path": "./tsconfig.lib.json" },
    { "path": "./tsconfig.test.json" }
  ]
}
