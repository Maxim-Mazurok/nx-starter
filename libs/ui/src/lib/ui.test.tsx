import { render } from '@testing-library/react';

import { Ui } from './ui';

describe('Ui', () => {
  const renderComponent = () => render(<Ui />);

  it('should render successfully', () => {
    const { baseElement } = renderComponent();

    expect(baseElement).toBeTruthy();
  });

  it('should match the snapshots', () => {
    const { asFragment } = renderComponent();

    expect(asFragment()).toMatchSnapshot();
  });
});
