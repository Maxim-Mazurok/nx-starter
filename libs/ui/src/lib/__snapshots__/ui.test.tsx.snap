// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ui should match the snapshots 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ dETfye"
    data-testid="idid"
  >
    <h1>
      Welcome to Ui!
    </h1>
  </div>
</DocumentFragment>
`;
